mm/2s2h/Rando/ActorBehavior/DmStk.cpp

```diff
@@ -77,4 +108,27 @@ void Rando::ActorBehavior::InitDmStkBehavior() {
         auto randoSaveCheck = RANDO_SAVE_CHECKS[RC_CLOCK_TOWER_ROOF_OCARINA];
         *should = !randoSaveCheck.obtained;
     });
+
+    COND_ID_HOOK(OnOpenText, 0x1C18, IS_RANDO && RANDO_SAVE_OPTIONS[RO_HINTS_OATH_TO_ORDER], ApplyOathHint);
```

Comment:
If his own text boxes won't work, document what this is, and how we're only overwriting it in the relevant scene